### DISCUSSION TITLE
Move all sitemaps / rss feeds into a better workflow

### DISCUSSION THREAD

**Discussion by MildTomato** [supabase/supabase#16847] [Contribute to Supabase]
# Chore

## Describe the chore

Perhaps move the RSS feed and Sitemap generation into a GitHub action.

Currently what is happening is .xml files are appearing in git commits as they are being generated on `npm run build` locally.

**Comment by MildTomato** [comment]
Example of PR needed to keep things up to date
https://github.com/supabase/supabase/pull/12463

**Comment by brandongregoryscott** [comment]
Hey @MildTomato, started taking a look at this - do the `apps/docs` and `apps/www` sites need to be built before running the `generate-sitemap.mjs` scripts? I'm not sure how those are built/deployed - not seeing any workflows that seem to run the build commands ~and it seems you need some environment variables set for it to build properly in CI.~

Edit: Nevermind about that - could've sworn I had a CI failure from attempting to build `apps/docs` or `apps/www` but I swapped it back to do a full build and seems to be running now.

I have a proof-of-concept `generate-sitemap` workflow working on my fork (PR example https://github.com/brandongregoryscott/supabase/pull/1/files) that leverages the [Create Pull Request](https://github.com/marketplace/actions/create-pull-request) Github Action to commit, push & PR changed XML files in the `apps/www` or `/apps/docs` directories.

**Reply by MildTomato** [reply]
[POC](https://github.com/supabase/supabase/pull/12514) that @brandongregoryscott did

**Reply by MildTomato** [reply]
[This was closed](https://github.com/supabase/supabase/pull/12514#issuecomment-1515723138) in favour of pursuing a solution keeping sitemaps and rss feeds out of version control, and have them created at build time
